tools/run_tests/xds_test_driver/framework/rpc/__init__.py

```diff
@@ -0,0 +1,95 @@
+# Copyright 2020 gRPC authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import logging
+import re
+from typing import Optional, ClassVar, Dict
+
+import grpc
+from google.protobuf import json_format
+import google.protobuf.message
+
+logger = logging.getLogger(__name__)
+
+# Type aliases
+Message = google.protobuf.message.Message
+
+
+class GrpcClientHelper:
+    channel: grpc.Channel
+    DEFAULT_CONNECTION_TIMEOUT_SEC = 60
+    DEFAULT_WAIT_FOR_READY_SEC = 60
+
+    def __init__(self, channel: grpc.Channel, stub_class: ClassVar):
+        self.channel = channel
+        self.stub = stub_class(channel)
+        # For better logging
+        self.service_name = re.sub('Stub$', '', self.stub.__class__.__name__)
+
+    def call_unary_when_channel_ready(
```

Comment:
I don't understand the intent of this helper method. It seems to just add wait_for_ready + a timeout to the call, but those are already args for grpc.UnaryUnaryMultiCallable. As far as I can tell this isn't actually doing anything to wait for the channel to be ready, e.g., querying the channel state - it's just adding two "fake" concepts (wait for ready sec, connection timeout sec) to make a deadline. (I realize some of this may be a port from the existing run_xds_tests.py).

I think having the helper is fine, but if what I wrote above is correct, it seems misnamed?

I'm also not sure what about this class justifies it being a class and extended elsewhere, versus just a helper method that can be invoked without inheritance, given that the main point of this class is just to set a deadline on an RPC?

Reply:
I'm pretty sure you're right, I'm still missing some basic gRPC concepts from the perspective a gRPC user :)
Would you have time for a GVC so I can as a couple of questions about where this fits conceptually?

Reply:
Sure, but also fine with leaving structure as-is (I am far from authoritative as far as Python style/structure goes) and just renaming to `call_unary_with_deadline`.